indico/modules/events/contributions/controllers/display.py

```diff
@@ -53,8 +55,11 @@ class RHContributionDisplayBase(RHDisplayEventBase):
 
     def _check_access(self):
         RHDisplayEventBase._check_access(self)
+        published = contribution_settings.get(self.event, 'published')
         if not self.contrib.can_access(session.user):
             raise Forbidden
+        if not published:
+            raise NotFound(_("The contributions of this event haven't been published yet."))
```

Comment:
let's be consistent with "haven't" vs "have not" (right now we have both in this PR)